From 60be1b77c27e803906c853e840e23ec25e51fe7c Mon Sep 17 00:00:00 2001
From: Joerg Roedel <jroedel@suse.de>
Date: Wed, 20 Aug 2014 16:32:53 +0200
Subject: [PATCH] PM / hibernate: Use put_page in free_image_page()
References: bnc#892004, bnc#890373
Patch-mainline: Never, fix for SLES-only PG_waiters patch

Use put_page to free image pages in the snapshot code.  This
makes sure that any stale PG_waiters bit is cleared before
the page is returned to the page allocator.

Signed-off-by: Joerg Roedel <jroedel@suse.de>
---
 kernel/power/snapshot.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/kernel/power/snapshot.c
+++ b/kernel/power/snapshot.c
@@ -145,7 +145,8 @@ static inline void free_image_page(void
 	if (clear_nosave_free)
 		swsusp_unset_page_free(page);
 
-	__free_page(page);
+	/* Use put_page here to make sure any stale page flags get cleared */
+	put_page(page);
 }
 
 /* struct linked_page is used to build chains of pages */
